### PR TITLE
[Refactor] Move enum ExpNotification into separated file

### PR DESCRIPTION
**Pull request by hayuna**
## What are the changes?
I've moved exp-notification enum into their own files inside src/data/enums

## Why am I doing these changes?
Moving enum to respective files
Avoiding hardcoded magic numbers and replace them with enums

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by Tempo-anon — COMMENTED** [review]
Looks good to me, but should battle-scene's `expParty` also be changed from an `integer` to the new enum?

File: src/phases.ts [hunk 1]
```diff
@@ -4289,20 +4290,20 @@ export class ShowPartyExpBarPhase extends PlayerPartyMemberPokemonPhase {
...
-        this.scene.partyExpBar.showPokemonExp(pokemon, exp.value, this.scene.expParty === 1, newLevel).then(() => {
+        this.scene.partyExpBar.showPokemonExp(pokemon, exp.value, this.scene.expParty === ExpNotification.ONLY_LEVEL_UP, newLevel).then(() => {
           setTimeout(() => this.end(), 800 / Math.pow(2, this.scene.expGainsSpeed));
         });
       } else {
         this.end();
       }
     } else if (this.scene.expGainsSpeed < 3) {
-      this.scene.partyExpBar.showPokemonExp(pokemon, exp.value, this.scene.expParty === 1, newLevel).then(() => {
+      this.scene.partyExpBar.showPokemonExp(pokemon, exp.value, false, newLevel).then(() => {
```
Comment (hayuna) [inline src/phases.ts:4306]: This condition will be always false, because `this.scene.expParty === 1` condition was checked in line 4295

**Comment by Tempo-anon** [comment]
Does the settings saving this selection still work? https://github.com/pagefaultgames/pokerogue/blob/20a3a4f60fe58a5fe929a51dff76a5db64080492/src/system/settings/settings.ts#L411

Just want to make sure so if you could test that then it should be all good to go

**Comment by hayuna** [comment]
Uploading video in progress

**Pull request merged**